tests/test_totem.lua

```diff
@@ -215,4 +219,52 @@ function tests.test_checkGradientsAcceptsGenericOutput()
     totem.nn.checkGradients(tester, mod, torch.randn(5, 5), 1e-6)
 end
 
+
+function tests.test_setUp()
+    tester:asserteq(test_name_passed_to_setUp, 'test_setUp')
+    for key, value in pairs(tester.tests) do
+        tester:assertne(key, '_setUp')
+    end
+end
+
+
+function tests.test_tearDown()
+    for key, value in pairs(tester.tests) do
+        tester:assertne(key, '_tearDown')
+    end
+end
+
+
+function tests._setUp(name)
+    test_name_passed_to_setUp = name
+    calls_to_setUp = calls_to_setUp + 1
+end
+
+
+function tests._tearDown(name)
+    calls_to_tearDown = calls_to_tearDown + 1
+end
+
+
 tester:add(tests):run()
+
+
+
+-- Additional tests to check that _setUp and _tearDown were called.
+local postTests = totem.TestSuite()
+
+
```

Comment:
perhaps run a small loop here to count the functions, instead of the magic '19' ?